src/util/transition/ngbTransition.spec.ts

```diff
@@ -0,0 +1,110 @@
+import {ngbRunTransition} from './ngbTransition';
+import createSpy = jasmine.createSpy;
+import {Component, ElementRef, ViewChild} from '@angular/core';
+import {TestBed} from '@angular/core/testing';
+
+function fadeFn({classList}: HTMLElement) {
+  classList.remove('ngb-test-show');
+}
+
+describe('ngbRunTransition', () => {
+
+  let element: HTMLElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({declarations: [TestComponent]});
+
+    const component = TestBed.createComponent(TestComponent);
+    component.detectChanges();
+    element = component.componentInstance.element.nativeElement;
+  });
+
+  it(`should run specified transition on an element`, (done) => {
+    element.classList.add('ngb-test-fade');
+    expect(window.getComputedStyle(element).opacity).toBe('1');
+
+    const nextSpy = createSpy();
+    const errorSpy = createSpy();
+
+    ngbRunTransition(element, fadeFn, {animation: true, running: 'continue'}).subscribe(nextSpy, errorSpy, () => {
+      expect(nextSpy).toHaveBeenCalledWith(undefined);
+      expect(element.classList.contains('ngb-test-show')).toBe(false);
+      expect(window.getComputedStyle(element).opacity).toBe('0');
+      expect(errorSpy).not.toHaveBeenCalled();
+      done();
+    });
+  });
```

Comment:
Just an idea, and I don't know if it's necessary or not, but here you test that everything is fine after that the asynchronous action is done. But you don't test the 'asynchronous nature' of the action.

I mean, if in ngbTransition the observer is resolved synchronously for any (wrong) reason (in this case, the expected transition is not run), this test will not fail, right ?

Just adding `expect(window.getComputedStyle(element).opacity).not.toBe('0');` after ngbRunTransition should ensure it.

What do you think ?

Reply:
Done, thanks